static/elements/chromedash-dialog.js

```diff
@@ -0,0 +1,159 @@
+import {LitElement, css, html} from 'lit-element';
+import '@polymer/iron-icon';
+
+// This is a simplfied version of chops-dialog:
+// https://source.chromium.org/chromium/infra/infra/+/main:appengine/monorail/static_src/elements/chops/chops-dialog/chops-dialog.js
+
+class ChromedashDialog extends LitElement {
+  static get properties() {
+    return {
+      opened: {type: Boolean, reflect: true},
+    };
+  }
+
+  static get styles() {
+    return css`
+      :host {
+        position: fixed;
+        z-index: 9999;
+        left: 0;
+        top: 0;
+        width: 100%;
+        height: 100%;
+        overflow: auto;
+        background-color: rgba(0,0,0,0.4);
+        display: flex;
+        align-items: center;
+        justify-content: center;
+      }
+      :host(:not([opened])), [hidden] {
+        display: none;
+        visibility: hidden;
+      }
+      :host([closeOnOutsideClick]),
+      :host([closeOnOutsideClick]) .dialog::backdrop {
+        /* TODO(zhangtiff): Deprecate custom backdrop in favor of native
+        * browser backdrop.
+        */
+        cursor: pointer;
+      }
+      .dialog {
+        background: none;
+        border: 0;
+        max-width: 90%;
+      }
+      .dialog-content {
+        /* This extra div is here because otherwise the browser can't
+        * differentiate between a click event that hits the dialog element or
+        * its backdrop pseudoelement.
+        */
+        box-sizing: border-box;
+        background: white;
+        padding: 1em 16px;
+        cursor: default;
+        box-shadow: 0px 3px 20px 0px hsla(0, 0%, 0%, 0.4);
+        width: var(--dialog-width-param);
+        max-width: var(--dialog-max-width-param, 100%);
+      }
+      #close-icon {
+        background: transparent;
+        border: 0;
+        float: right;
+        cursor: pointer;
+      }
+
+    `;
+  }
+
+  render() {
+    return html`
+      <dialog class="dialog" role="dialog" @cancel=${this._cancelHandler}>
```

Comment:
(not sure) should the `opened` property in this class also handle the native dialog `open` attribute? (and I guess that could simplify some manual handling in this component)

Reply:
I see what you mean.  Looking at the original [chops-dialog](https://source.chromium.org/chromium/infra/infra/+/main:appengine/monorail/static_src/elements/chops/chops-dialog/chops-dialog.js) class, I think that it did not use something like
`<dialog ?open=${this.opened}>`
because calling `dialog.showModal()` is somehow better than setting the `open` attribute.  However, while reading the MDN docs now, I don't see exactly what would make it better, other than being the newer way to do it.  So, I am just sticking with the existing logic.